npm/design-system/src/components/FileExplorer/FileExplorer.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import { makeFileHierarchy } from './helpers/makeFileHierarchy'
+import { FileExplorerProps, FileTreeProps, FileLike, FolderOrFile } from './types'
+import styles from './FileExplorer.module.scss'
+import { InlineIcon } from '@iconify/react'
+import javascriptIcon from '@iconify/icons-vscode-icons/file-type-js-official'
+import typescriptIcon from '@iconify/icons-vscode-icons/file-type-typescript-official'
+import reactJs from '@iconify/icons-vscode-icons/file-type-reactjs'
+import reactTs from '@iconify/icons-vscode-icons/file-type-reactts'
+import folderClosed from '@iconify/icons-vscode-icons/default-folder'
+import folderOpen from '@iconify/icons-vscode-icons/default-folder-opened'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import cs from 'classnames'
+
+library.add(fas)
+
+const icons: Record<string, any> = {
+  js: { icon: javascriptIcon },
+  ts: { icon: typescriptIcon },
+  tsx: { icon: reactTs },
+  jsx: { icon: reactJs },
+  folderOpen: { icon: folderOpen },
+  folderClosed: { icon: folderClosed },
+}
+
+const getExt = (path: string) => {
+  const extensionMatches = path.match(/(?:\.([^.]+))?$/)
+
+  return extensionMatches ? extensionMatches[1] : ''
+}
+
+export const FileExplorer: React.FC<FileExplorerProps> = (props) => {
+  return (<nav className={cs(props.className || '', styles.nav)}><FileTree files={makeFileHierarchy(props.files) || []} isSelected={props.isSelected || (() => {})} onClick={props.onClick || (() => {})}></FileTree></nav>)
+}
```

Comment:
Do we need all the `|| () => {}`? Can we just make the props nullable?

Reply:
And even if not, you should use `??`

Reply:
I don't think we can use ?? with our version of ts? I'm not sure.

Reply:
We are not using ts for compilation it is extremely bad for this (if we are we should stop). Babel will handle this much better because we are still using it.

Reply:
> ?? is a stage 3 so preset-env will be enough and I think it will work out of the box

https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining if not

Reply:
However, @dmtrKovalenko, if the TS version we're using doesn't support it, we'll get errors from the LSP